crates/turborepo-auth/src/auth/login.rs

```diff
@@ -267,6 +293,60 @@ mod tests {
         }
     }
 
+    #[async_trait]
```

Comment:
Perhaps out of scope for this PR but as of rust 1.75 we don't need async_trait any more. If you need to specify things like `Send` (the default for `async_trait` traits) you can use return position `impl` instead.

```rust
async fn get_artifact(...) -> Result<Option<Response>, turborepo_api_client::Error>
// or
fn get_artifact(...) -> impl Future<Output=Result<Option<Response>, turborepo_api_client::Error>> + Send
```

The side effect being no more `Pin<Box<dyn Future>>` :)

See here: https://github.com/vercel/turbo/blob/main/crates/turborepo-repository/src/discovery.rs#L43-L57